File: 2]_Algorithms/basic algo sorting in java/selectionsort.java
public class selectionsort {
    public static void selection(int arr[]){
        for(int i=0;i<arr.length-1;i++){// array.length= 5 --> 0 to 4
            int smallestnumber=i;
            for(int j=i+1;j<arr.length;j++){ // 1 to 5
                if(arr[smallestnumber]>arr[j]){ // arr[smallestnumber]<arr[j]= desending order
                    smallestnumber=j; // we got smallest number and have to place in 1st indrx that is i
                }}
                //outer loop swap 
                int temp=arr[smallestnumber];
                arr[smallestnumber]=arr[i];
                arr[i]=temp; // swapped i & smallestnumber


            
        }
    }
    public static void main(String args[]){
        int arr[]= {5,2,3,1,4};
        selection(arr);// fcall 
        // print array
        for(int i=0;i<arr.length;i++){
            System.out.print( arr[i] +" ");
        }

    }
    
}

File: 2]_Algorithms/2]_Sorting/Sorting in C++/Selectionsort.cpp
#include <iostream>
using namespace std;

void swap(int *x, int *y)
{
    int temp = *x;
    *x = *y;
    *y = temp;
}
void selectionsort(int arr[], int n)
{
    for (int i = 0; i < n - 1; i++)
    {
        int min_idx = i;
        for (int j = i + 1; j < n; j++)
        {
            if (arr[j] < arr[min_idx])
                min_idx = j ;
        }
        if (arr[min_idx] != arr[i])
        {
            swap(&arr[min_idx], &arr[i]);
        }
    }
}
int main()
{
    int n;
    cout << "Enter size of Array\n";
    cin >> n;
    int arr[n];

    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }
    selectionsort(arr, n);
    // printing array
    for (int i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
}